x-pack/plugin/ml/log-structure-finder/src/main/java/org/elasticsearch/xpack/ml/logstructure/LogFileStructure.java

```diff
@@ -0,0 +1,614 @@
+            this.numMessagesAnalyzed = numMessagesAnalyzed;
+            return this;
+        }
+
+        public Builder setSampleStart(String sampleStart) {
+            this.sampleStart = Objects.requireNonNull(sampleStart);
+            return this;
+        }
+
+        public Builder setCharset(String charset) {
+            this.charset = Objects.requireNonNull(charset);
+            return this;
+        }
+
+        public Builder setHasByteOrderMarker(Boolean hasByteOrderMarker) {
+            this.hasByteOrderMarker = hasByteOrderMarker;
+            return this;
+        }
+
+        public Builder setFormat(Format format) {
+            this.format = Objects.requireNonNull(format);
+            this.separator = format.separator();
+            return this;
+        }
+
+        public Builder setMultilineStartPattern(String multilineStartPattern) {
+            this.multilineStartPattern = multilineStartPattern;
+            return this;
+        }
+
+        public Builder setExcludeLinesPattern(String excludeLinesPattern) {
+            this.excludeLinesPattern = excludeLinesPattern;
+            return this;
+        }
+
+        public Builder setInputFields(List<String> inputFields) {
+            this.inputFields = inputFields;
+            return this;
+        }
+
+        public Builder setHasHeaderRow(Boolean hasHeaderRow) {
+            this.hasHeaderRow = hasHeaderRow;
+            return this;
+        }
+
+        public Builder setShouldTrimFields(Boolean shouldTrimFields) {
+            this.shouldTrimFields = shouldTrimFields;
+            return this;
+        }
+
+        public Builder setSeparator(Character separator) {
+            this.separator = separator;
+            return this;
+        }
+
+        public Builder setGrokPattern(String grokPattern) {
+            this.grokPattern = grokPattern;
+            return this;
+        }
+
+        public Builder setTimestampField(String timestampField) {
+            this.timestampField = timestampField;
+            return this;
+        }
+
+        public Builder setTimestampFormats(List<String> timestampFormats) {
+            this.timestampFormats = timestampFormats;
+            return this;
+        }
+
+        public Builder setNeedClientTimezone(boolean needClientTimezone) {
+            this.needClientTimezone = needClientTimezone;
+            return this;
+        }
+
+        public Builder setMappings(Map<String, Object> mappings) {
+            this.mappings = Objects.requireNonNull(mappings);
+            return this;
+        }
+
+        public Builder setExplanation(List<String> explanation) {
+            this.explanation = Objects.requireNonNull(explanation);
+            return this;
+        }
+
+        @SuppressWarnings("fallthrough")
+        public LogFileStructure build() {
+
+            if (numLinesAnalyzed <= 0) {
+                throw new IllegalArgumentException("Number of lines analyzed must be positive.");
```

Comment:
Another inconsistency here is whether the error messages contain a natural description of the field vs. the actual field name.

Reply:
I think it's academic in this case, because it's our code supplying the values rather than the end user.  So this check won't trigger as a result of a user calling a REST endpoint unless there's a bug in our code.